### PR TITLE
docs(Effects): document defining effects as function

### LINKED ISSUE #1198: Effects: Document effects as functions
## I would be willing to submit a PR for the docs :heart:

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

Effects can be defined in two ways but only one way is documented.

Property Assignment
```
@Effect()
this.effect$ = this.actions$.pipe(ofType('SOME_ACTION'));
```

Using a function that is called without arguments and returns the effect.
```
@Effect()
effect$() {
  return this.actions$.pipe(ofType('SOME_ACTION'));
}
```

Defining effects using a function allows you to use defaults that can be overridden during testing. This could be a useful pattern in providing test schedulers or custom values during testing.

```
@Effect()
effect$(waitTime = 3000, scheduler: Scheduler | undefined = undefined) {
  return this.actions$.pipe(ofType('SOME_ACTION')).pipe(delay(waitTime, scheduler));
}
```

### PR DESCRIPTION
**Pull request by timdeschryver**
Closes #1198

### CODE REVIEW

File: docs/effects/testing.md [hunk 1]
```diff
@@ -104,3 +104,44 @@ describe('My Effects', () => {
   });
 });
 ```
+
+### Effects as functions
+
+Sometimes testing a more complex effect becomes tricky, for example the effect may depend on external functionality like RxJS schedulers, may have an high `interval`/`throttleTime`/`debounceTime`, or ... you probably get the idea.
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply docs/effects/testing.md]: "... you probably get the idea" is too vague. We should at least describe what kind of operators these are.

File: docs/effects/testing.md [hunk 2]
```diff
@@ -104,3 +104,43 @@ describe('My Effects', () => {
...
+
+```ts
+const actual = effects.search$({
+  debounce: 30,
+  scheduler: getTestScheduler(),
+});
+expect(actual).toBeObservable(expected);
+```
+
+Doing this has the extra benefit of hiding implementation details, making your tests less prone to break due to implementation details changes. If you would change the default `debounceTime` you wouldn't have to change your tests, these would still pass.
```
Comment (timdeschryver) [inline docs/effects/testing.md]: Should mention this?

File: docs/effects/testing.md [hunk 3]
```diff
@@ -104,3 +104,43 @@ describe('My Effects', () => {
   });
 });
 ```
+
+### Effects as functions
+
+Effects can be defined as functions as well as variables. Defining an effect as a function allows you to define default values while having the option to override these variables during tests. This without breaking the functionality in the application.
+
+For example, if we would have an effect that handles a typeahead we may want to debounce the user's input:
```
Comment (brandonroberts) [inline docs/effects/testing.md]: Update to describe the example. 


"The following example effect debounces the user input into from a search action."

Comment (timdeschryver) [reply docs/effects/testing.md]: Thanks for being so patient with me 😅

Comment (brandonroberts) [reply docs/effects/testing.md]: No problem at all 👍

**Pull request merged**